eureka/S5_lightcurve_fitting/models/BatmanModels.py

```diff
@@ -0,0 +1,210 @@
+        self.parameters = kwargs.get('parameters')
+
+        # Generate parameters from kwargs if necessary
+        if self.parameters is None:
+            self.parameters = Parameters(**kwargs)
+        
+        # Set parameters for multi-channel fits
+        self.longparamlist = kwargs.get('longparamlist')
+        self.nchan = kwargs.get('nchan')
+        self.paramtitles = kwargs.get('paramtitles')
+
+    def eval(self, **kwargs):
+        """Evaluate the function with the given values"""
+        # Get the time
+        if self.time is None:
+            self.time = kwargs.get('time')
+        
+       #Initialize model
+        bm_params = batman.TransitParams()
+        pc_params = {'A':0., 'B':0., 'C':0., 'D':0.}
+
+        # Set all parameters
+        lcfinal=np.array([])
+        for c in np.arange(self.nchan):
+            # Set all parameters
+            for index,item in enumerate(self.longparamlist[c]):
+                if item in pc_params.keys():
+                    pc_params[self.paramtitles[index]] = self.parameters.dict[item][0]
+                else:
+                    setattr(bm_params, self.paramtitles[index], self.parameters.dict[item][0])
+
+            bm_params.limb_dark = 'uniform'
+            bm_params.u = []
+
+            m_transit = None
+            if not np.any(['t_secondary' in key for key in self.longparamlist[c]]):
+                # If not explicitly fitting for the time of eclipse, get the time of eclipse from the time of transit, period, eccentricity, and argument of periastron
+                m_transit = batman.TransitModel(bm_params, self.time, transittype='primary')
+                t_secondary = m_transit.get_t_secondary(bm_params)
+            else:
+                t_secondary = self.parameters.dict['t_secondary'][0]
+
+            if self.parameters.dict['ecc'] == 0.:
+                #the planet is on a circular orbit
+                t    = self.time - t_secondary
+                freq = 2.*np.pi/self.parameters.dict['per']
+                phi  = (freq*t)
+            else:
+                if m_transit is not None:
+                    # Avoid overhead of making a new transit model if avoidable
+                    m_transit = batman.TransitModel(bm_params, self.time, transittype='primary')
+                anom = m_transit.get_true_anomaly()
+                w = self.parameters.dict['w'][0]
+                #the planet is on an eccentric orbit
+                phi  = anom + w*np.pi/180. + np.pi/2.
+            
+            #calculate the phase variations
+            if pc_params['C']==0. and pc_params['D']==0.:
+                #Skip multiplying by a bunch of zeros to speed up fitting
+                phaseVars = 1. + pc_params['A']*(np.cos(phi)-1.) + pc_params['B']*np.sin(phi)
+            else:
+                phaseVars = 1. + pc_params['A']*(np.cos(phi)-1.) + pc_params['B']*np.sin(phi) + pc_params['C']*(np.cos(2.*phi)-1.) + pc_params['D']*np.sin(2.*phi)
```

Comment:
From what I can tell, this implementation doesn't remove the curvature during secondary eclipse.  While this is a good start, we'll want to implement more advanced versions down the road.

Reply:
By this you mean that it doesn't model eclipse mapping type observations? That is definitely true and a limitation of this quick and simple first attempt

Reply:
Or are you meaning that there would be phase curvature during the bottom of eclipse? I'll need to double check that, but I'm pretty sure the eclipse function goes to zero during the bottom of eclipse so that there doesn't end up being any curvature during the flat part of the eclipse. That's what SPCA does, but I haven't checked to make sure I carried that part over

Reply:
Oops, it does not - working on that now